srcalc/src/main/java/gov/va/med/srcalc/vista/vistalink/VistaLinkProcedureCaller.java

```diff
@@ -75,30 +84,55 @@ public VistaLinkVistaDao(final String division)
         }
     }
     
-    public VistaPerson loadVistaPerson(final String duz)
+    @Override
+    public String getDivision()
     {
-        fLogger.debug("Loading VistaPerson for duz {}.", duz);
+        return fDivision;
+    }
 
+    @Override
+    public List<String> doRpc(final String duz, final String rpcName, final String... args)
+    {
         // This is the RPC context for all ASRC RPCs. (This value is determined
         // by VistA.)
         final String rpcContext = "SR ASRC";
 
         final VistaLinkDuzConnectionSpec cs =
                 new VistaLinkDuzConnectionSpec(fDivision, duz);
-        
+
         try
         {
+            fLogger.debug("About to call remote procedure \"{}\"", rpcName);
             final RpcRequest req = RpcRequestFactory.getRpcRequest(
-                    rpcContext, "SR ASRC USER");
+                    rpcContext, rpcName);
+            
+            // Set the arguments.
+            for (int i = 0; i < args.length; ++i)
+            {
+                // VistA starts counting at 1, not 0.
+                final int vistaParamIndex = i + 1;
```

Comment:
At least here it's all encapsulated in `VistaLinkProcedureCaller` so you don't have to!